avalancheup-aws/src/apply/mod.rs

```diff
@@ -1765,9 +1765,12 @@ pub async fn execute(log_level: &str, spec_file_path: &str, skip_prompt: bool) -
 
                 if spec.machine.ip_mode == *"elastic" {
                     log::info!("using elastic IPs... wait more");
-                    let mut outs: Vec<Address>;
+                    #[allow(unused_assignments)]
+                    let mut eips_out: Vec<Address> = Vec::new();
+                    let max_retry = 10;
                     loop {
-                        outs = regional_ec2_manager
+                        let mut retry_count = 0;
+                        let outs = match regional_ec2_manager
```

Comment:
Suggest using https://docs.rs/backoff/latest/backoff/, which uses an exponential backoff as well as adding some randomness, and simplifies your code down significantly:
```suggestion
                        let backoff = ExponentialBackoff::default();
                        let outs = retry(|| regional_ec2_manager.describe_eips_by_tags(HashMap::from([
                                (String::from("Id"), spec.id.clone()),
                                (
                                ),
                            ]))
                            .await, backoff).unwrap();
```

Reply:
I think this is a good idea but I suggest that it should be done as part of a wider effort in the crate rather than a one off. I will open an issue to track using `backoff` in the project.

Reply:
Actually, I'll try doing it here too. It doesn't look so bad to implement